### ISSUE TITLE
Consider removing part of requirements checks for podman installation starting with 5.2.3

### ISSUE THREAD

**Issue by odockal** [containers/podman-desktop#9159]
### Is your enhancement related to a problem? Please describe

Right now we have requirements check present on podman desktop side. We have also issue with it (https://github.com/containers/podman-desktop/issues/9158). But podman 5.2.3 installer offers to enable wsl and hyperv as per user choice. In podman desktop we are having a redundant functionality which also the installer offers. 

### Describe the solution you'd like

Consider removing some of the requirements checks for podman installation since the podman installer have sufficient functionality incorporated. 

We might also consider keep the check once the installation is done but none of the required features is installed and thus user cannot really create a machine. Revamping would be great idea. I would think of a moving the checks into podman machine create page, where it might work before user actually creates a machine. 

### Additional context

Related to https://github.com/containers/podman-desktop/issues/9158.

**Comment by odockal** [comment]
We was multiple issues that are related and could be fixed/closed if we re-do requirements check on Podman Desktop side.

* https://github.com/podman-desktop/podman-desktop/issues/10462
* https://github.com/podman-desktop/podman-desktop/issues/7414
* https://github.com/podman-desktop/podman-desktop/issues/13183
* https://github.com/podman-desktop/podman-desktop/issues/11791
* https://github.com/podman-desktop/podman-desktop/issues/12173
* https://github.com/podman-desktop/podman-desktop/issues/12499
* https://github.com/podman-desktop/podman-desktop/issues/7827

**Comment by odockal** [comment]
Please consider this issue again. Podman 6 is incoming and we would still need to do some investigation. Let's consider leaving podman installer to setup necessary providers.

**Comment by jiridostal** [comment]
First step for investigation: Identify what needs to be removed/is redundant with podman installer.

**Comment by gastoner** [comment]
## Podman Windows Installer — Evolution of Checks

### MSI Installer

The MSI installer runs when you install the Podman CLI binary onto Windows.

| Version | Behavior | Reference |
|---------|----------|-----------|
| < 5.1 | Installs binaries only. WSL-only checkbox to install WSL during installation. | — |
| **5.1** (May 2024) | Adds **radio buttons** to choose WSL or Hyper-V. Checkbox to install WSL/Hyper-V if missing. Creates `99-podman-machine-provider.conf`. | [PR #22506](https://github.com/containers/podman/pull/22506), [PR #22836](https://github.com/containers/podman/pull/22836) |
| **5.4** (Nov 2024) | WSL/Hyper-V install checkbox **unchecked by default**. Fixed bug where update reinstalled the provider. | [PR #24624](https://github.com/containers/podman/pull/24624) |
| **5.5** (Feb 2025) | **Removed** the option to install WSL/Hyper-V entirely from the GUI. Installer only selects the provider and installs binaries. | [PR #25237](https://github.com/containers/podman/pull/25237) |
| **5.5+** (May 2025) | Removed dead `check.c` / DLL code that was no longer used. | [PR #26023](https://github.com/containers/podman/pull/26023) |
| **6.0** (Mar 2026) | Same as 5.5+ — installer installs binaries + provider config only. Drops Windows 10 support. Supports user-scope and machine-scope installation. | [podman6hld.md](https://github.com/containers/podman/blob/main/contrib/design-docs/podman6hld.md), [PR #27648](https://github.com/containers/podman/pull/27648) |

### `podman machine init`

This runs when the user creates a Podman machine (VM).

| Version | Behavior | Reference |
|---------|----------|-----------|
| < 5.5 | **Assumes** WSL is installed. If WSL import fails, shows a cryptic error. No real detection or auto-install. | [Issue #25234](https://github.com/containers/podman/issues/25234) |
| **5.5** (Feb 2025) | Adds WSL detection via `wsl --status`. If WSL is missing, **attempts** auto-install (DISM to enable VirtualMachinePlatform + WSL features, prompts reboot). **But this flow was broken** — exit code 3010 not handled, lock issues on re-exec. | [PR #25522](https://github.com/containers/podman/pull/25522), [Issue #25523](https://github.com/containers/podman/issues/25523) |
| **5.6** (Jun 2025) | **Fixes** the WSL auto-install flow. New detection: tries the actual WSL import, checks for HCS errors, then enables features via DISM. Handles reboot properly. | [PR #26201](https://github.com/containers/podman/pull/26201) |
| **6.0** (Mar 2026) | Same working auto-install flow. Provider can be specified via `--provider` flag. Machines from different providers can coexist (no more strict provider locking). Windows 10 dropped. | [podman6hld.md](https://github.com/containers/podman/blob/main/contrib/design-docs/podman6hld.md) |

---

## Podman 6 Plans for Windows

Source: [Podman 6 High Level Design](https://github.com/containers/podman/blob/main/contrib/design-docs/podman6hld.md)

| Area | What changes |
|------|-------------|
| **Windows 10** | Dropped. Windows 11 only (Microsoft EOL'd Win10 in Oct 2025). |
| **Provider flexibility** | No longer locked to one provider. Users can have WSL and Hyper-V machines side-by-side. `podman machine stop myWSLmachine` works even if Hyper-V is the default. |
| **`--provider` flag** | `podman machine init --provider wsl` or `--provider hyperv` to override the default. |
| **Machine listing** | `podman machine list` shows all machines across all providers by default. |
| **Config files** | Major rework of `containers.conf` — client vs server separation. Machine provider config only applies on the client (Windows/Mac), not inside the VM. |
| **WSL updates** | Acknowledges WSL is a special case (can't use OCI/ostree). Future direction unclear — may allow `dnf update` or find a unified approach. |

---

## Overlap Matrix: Podman Desktop vs Podman

| Check | Podman Desktop | MSI Installer (6.0) | `machine init` (6.0) | Overlaps? |
|-------|---------------|--------------------|--------------------|-----------|
| Windows 64-bit | `WinBitCheck` | — | — | **No** — PD only |
| Windows version >= Build 19043 | `WinVersionCheck` | — | Checks `winVersionAtLeast(10, 0, 19041)` | Slight — PD uses higher min build |
| RAM >= 5 GB | `WinMemoryCheck` | — | — | **No** — PD only |
| VirtualMachinePlatform enabled | `VirtualMachinePlatformCheck` | — | Enables via DISM if missing | **Yes** — Podman handles it |
| WSL installed | `WSL2Check` | — | Detects + auto-installs if missing | **Yes** — Podman handles it |
| WSL version >= 1.2.5 | `WSLVersionCheck` | — | Checks via `wsl --status` | **Yes** — Podman handles it |
| Hyper-V installed | `HyperVInstalledCheck` | — | — | Partial — MSI provider selection |
| Hyper-V running | `HyperVRunningCheck` | — | — | **No** — PD only |
| User is admin | `UserAdminCheck` | Requests elevation if needed | Prompts elevation for WSL install | **Yes** — Podman handles it |
| PD running elevated | `PodmanDesktopElevatedCheck` | — | — | **No** — PD-specific |
| Podman >= 5.2.0 for Hyper-V | `HyperVPodmanVersionCheck` | Shows Hyper-V only on 5.2+ | — | **Yes** — MSI handles it |
| Provider selection | `isWSLEnabled` / `isHyperVEnabled` | Radio buttons in GUI | `--provider` flag, env var, config | **Yes** — MSI and CLI both handle it |
| Reboot detection | `WSL2Check` (checks `WSL_E_WSL_OPTIONAL_COMPONENT_REQUIRED`) | — | Handles DISM exit code 3010, prompts reboot | **Yes** — Podman handles it |

---

## Summary

### Current situation (Podman 5.8+ / 6.0)

- The MSI installer **no longer** installs WSL or Hyper-V — this was **removed** in 5.5
  ([PR #25237](https://github.com/containers/podman/pull/25237)) because it caused regressions
  and was hard to maintain and test.
- `podman machine init` **detects and auto-installs** WSL if missing — fixed in 5.6
  ([PR #26201](https://github.com/containers/podman/pull/26201)).
- Podman Desktop's virtualization preflight checks overlap heavily with what
  `podman machine init` now handles.

### What Podman Desktop should keep

The following checks have **no equivalent** in Podman upstream and provide early, cheap
feedback before the user starts the install:

- **WinBitCheck** — 64-bit OS (x64 or arm64)
- **WinVersionCheck** — Windows 10 Build 19043+ (or Windows 11)
- **WinMemoryCheck** — at least 5 GB RAM

### What Podman Desktop can remove from preflight

For bundled Podman >= 5.6.0, the virtualization checks can be removed from the
pre-installation preflight since Podman handles them at machine creation time:

- `VirtualMachinePlatformCheck`
- `WSL2Check`
- `WSLVersionCheck`
- `HyperVInstalledCheck`
- `HyperVRunningCheck`
- `UserAdminCheck`
- `PodmanDesktopElevatedCheck`
- `HyperVPodmanVersionCheck`

**Comment by gastoner** [comment]
Latest update: 
Some checks like WSL, Virtual Machine Platform or HyperV are no longer needed to install latest Podman versions.
Due to this we can move those checks to 'optional' ones (error ❗ ->  warnings ⚠️ ) since Podman will handle those during `podman machine init` command. So we would just show the user that those checks are not passing, with all the buttons to enable those (as has been done previously), but we wont require those checks to finish with ✅ .

**Comment by gastoner** [comment]
We might also drop some of the checks entirely

**Comment by gastoner** [comment]
@odockal @jeffmaury @slemeur Thinking what would be the best option here, if to switch to warnings, or removing the checks WDYT?

**Comment by slemeur** [comment]
Agree to update the preflight checks. 
I believe it is still relevant to show the detection of the different underlying tools if they are already. But we should check the wording. For instance, if we don't detect WSL, we could say "will be installed as part of the installation flow".

**Comment by gastoner** [comment]
@slemeur 
`Virtual Machine Platform should be enabled to be able to run Podman.` -> `Virtual Machine Platform should be enabled to run Podman using WSL. Podman will try to enable it automatically when you create a machine (restart required).`

`WSL2 is not installed.` -> `WSL2 is not installed. Use the "Install WSL2" button, or Podman will attempt to install it automatically when you create a machine (restart required).`

`WSL2 is not installed or you do not have permissions to run WSL2.` -> `WSL2 is not installed or you do not have permissions to run WSL2. Podman will still attempt to install it automatically when you create a machine, prompting for admin rights and a restart.`

WDYT?

**Comment by jeffmaury** [comment]
> [@slemeur](https://github.com/slemeur) `Virtual Machine Platform should be enabled to be able to run Podman.` -> `Virtual Machine Platform should be enabled to run Podman using WSL. Podman will try to enable it automatically when you create a machine (restart required).`
> 
> `WSL2 is not installed.` -> `WSL2 is not installed. Use the "Install WSL2" button, or Podman will attempt to install it automatically when you create a machine (restart required).`
> 
> `WSL2 is not installed or you do not have permissions to run WSL2.` -> `WSL2 is not installed or you do not have permissions to run WSL2. Podman will still attempt to install it automatically when you create a machine, prompting for admin rights and a restart.`
> 
> WDYT?

Not sure it is done at machine creation time but at podman installation time ? Also we need to check the combination with WSL vs HyperV
